iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -799,7 +799,7 @@ public override async Task SendMethodResponseAsync(DirectMethodResponse methodRe
             cancellationToken.ThrowIfCancellationRequested();
             EnsureValidState();
 
-            var message = new Message(methodResponse.Payload)
+            var message = new Message((byte[])methodResponse.Payload)
```

Comment:
Do we need to worry about exception on explicit type casting?

Reply:
Message constructor expects byte[] as the parameter, In the latest commit, I have also updated the code to explicitly convert the payload to byte[].